patches/api/0296-Datapack-API.patch

```diff
@@ -42,16 +43,41 @@ index 0000000000000000000000000000000000000000..7b2ab0be10a21e0496ad1d485ff8cb2c
 +    }
 +
 +}
+diff --git a/src/main/java/io/papermc/paper/datapack/DatapackFileNotFoundException.java b/src/main/java/io/papermc/paper/datapack/DatapackFileNotFoundException.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..52e5a014b6640f3ea4541af42bdab819ceae7352
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/datapack/DatapackFileNotFoundException.java
+@@ -0,0 +1,15 @@
++package io.papermc.paper.datapack;
++
++import org.jetbrains.annotations.NotNull;
++
++import java.io.File;
++import java.io.FileNotFoundException;
++
++/**
++ * Thrown when the server can't load a datapack because a file is missing.
++ */
++public class DatapackFileNotFoundException extends FileNotFoundException {
++    public DatapackFileNotFoundException(@NotNull File packSource, @NotNull String resource) {
++        super(String.format("'%s' in DataPack '%s'", resource, packSource));
++    }
++}
```

Comment:
Do we need a custom exception? All its doing is a custom string.format, can't that just be done in the FileNotFoundException ctor?